src/textual_fspicker/parts/directory_navigation.py

```diff
@@ -469,6 +480,8 @@ def _on_option_list_option_selected(self, event: OptionList.OptionSelected) -> N
         assert isinstance(event.option, DirectoryEntry)
         # If the use has selected a directory...
         if is_dir(event.option.location):
+            if self.double_click_directories and not self.last_event_doubleclick:
```

Comment:
I think the code dealing with double-clicks, in this method, might need something else. In testing this I've found that it's possible to get unintended navigation with 3 clicks. That is, if you double click on a directory at the top of a list, and then click once more, it'll appear to be a double click on whatever directory is in that location after the first navigation has taken place.

I suspect it's as simple as `last_event_doubleclick` needing setting to `False` again, or perhaps some more work needs doing revolving around the use of `event.chain`?

Reply:
I opted for option 1, since it seems more robust and felt good during my brief test. If other users complain, we could reconsider changing the implementation.

Reply:
Yup, agreed; playing with it here it feels right to me too. The more I play with this the more I like the PR in general too.